js/render-comments.ts

```diff
@@ -1,40 +1,68 @@
 import { PhotoComment } from './contracts/common';
 import { findBEMElement, findTemplate } from './utils';
 
+
 const commentsCount = document.querySelector<HTMLSpanElement>('.comments-count');
+const commentsCurrentCount = document.querySelector<HTMLSpanElement>('.comments-current-count');
 const commentCounter = document.querySelector<HTMLDivElement>('.social__comment-count');
-const commentLoaderButton = document.querySelector<HTMLButtonElement>('.comments-loader');
+const commentLoaderButton = document.querySelector<HTMLButtonElement>('.social__comments-loader');
 const commentsListFragment = document.createDocumentFragment();
 const commentTemplate = findTemplate<HTMLLIElement>('comment');
 const commentsList = document.querySelector<HTMLUListElement>('.social__comments');
 
-if (!commentCounter || !commentLoaderButton || !commentsList || !commentsCount) {
-	throw new Error('Critical elements for Comments were not found');
+const NUMBER_OF_UPLOADED_COMMENTS = 5;
+
+if (!commentCounter || !commentLoaderButton || !commentsList || !commentsCount || !commentsCurrentCount || !commentsList) {
+	throw new Error('Critical elements for comments were not found');
 }
 
-commentLoaderButton.hidden = true;
-commentCounter.hidden = true;
+let shownCommentsAmount = 0;
+
+const getShownCommentsAmount = (shownCommentsCount: number, totalCommentsAmount: number): number => {
+	const differenceOfTwoNumbers = totalCommentsAmount - shownCommentsCount;
+	if (differenceOfTwoNumbers > 0) {
+		shownCommentsAmount += Math.min(NUMBER_OF_UPLOADED_COMMENTS, differenceOfTwoNumbers);
+	}
+
+	return shownCommentsAmount;
+};
 
 const renderComments = (comments: PhotoComment[]) => {
-	commentsCount.textContent = comments.length.toString();
+	const totalCommentsAmount = comments.length;
+	commentsCount.textContent = totalCommentsAmount.toString();
+	const currentShownCommentsAmount = getShownCommentsAmount(shownCommentsAmount, totalCommentsAmount);
+	commentsCurrentCount.textContent = currentShownCommentsAmount.toString();
+	commentLoaderButton.classList.remove('hidden');
+
+	return () => {
+		const currentPartToRender = comments.slice(shownCommentsAmount, shownCommentsAmount + NUMBER_OF_UPLOADED_COMMENTS);
+
+		currentPartToRender.forEach((comment: PhotoComment) => {
```

Comment:
Конкретно тут оно и так считывает тип
```suggestion
		currentPartToRender.forEach((comment) => {
```

Но вообще можно эту функцию из `forEach` вынести на верхний уровень модуля, она достаточно чистая для этого. Тогда и входной тип пригодится